### PR TITLE
Structural Refactor

### PR DESCRIPTION
**Pull request by mikesizz**
This PR is intended to clean up the API file hierarchy as well as bringing some small QoL improvements.

- Move all route handlers to their own files.
- Remove unused code.
- Add nodemon support for local development.
- Add prettier support.
- Add basic eslint support.
- Group utility methods together.
- Minor Fixes
  - Correct express support for Promises.
  - Minor performance improvements - unnecessary iterations, Promise.all, etc.

### CODE REVIEW

**Comment by mikesizz** [comment]
At the time of this authoring the master and develop branch were not up-to-date with each other. Fixes existed in master that were not in develop. I also suspect some work is missing from master, or the live site is rolled back to some unknown (to me) version, as it has some issues (called out in comments) that don't exist on the live site.

File: src/lib/conseil.js [hunk 1]
```diff
@@ -0,0 +1,1101 @@
...
+    mintOperationQuery,
+    'kind',
+    conseiljs.ConseilOperator.EQ,
+    ['transaction']
+  )
+  mintOperationQuery = conseiljs.ConseilQueryBuilder.addPredicate(
+    mintOperationQuery,
+    'timestamp',
+    conseiljs.ConseilOperator.BETWEEN,
+    [d.getTime(), max_time]
```
Comment (mikesizz) [inline src/lib/conseil.js:771]: /random seems to not work as it passes in zero. This seems to be an invalid range. No results returned locally.

File: src/lib/conseil.js [hunk 2]
```diff
@@ -0,0 +1,1101 @@
...
+    balance = balanceResult[0]['balance'] // TODO: consider bigNumber here, for the moment there is no reason for it
+  } catch (error) {
+    console.log(
+      `getTezBalanceForAddress failed for ${JSON.stringify(
+        accountQuery
+      )} with ${error}`
+    )
+  }
+
+  return balance
```
Comment (mikesizz) [inline src/lib/conseil.js:416]: This is undefined in master.

File: src/lib/router/readHdaoFeed.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const { getObjktById, paginateFeed } = require('utils')
+
+module.exports = async function readHdaoFeed(req, res) {
+  const rawFeed = await conseil.hDAOFeed()
+  const sortedFeed = _.orderBy(rawFeed, ['hDAO_balance'], ['desc'])
+  const paginatedFeed = paginateFeed(sortedFeed, 0)
+
+  res.json({
+    result: await Promise.all(
+      paginatedFeed.map(async (objkt) => await _mergeHdao(objkt))
```
Comment (mikesizz) [inline src/lib/router/readHdaoFeed.js:15]: I've tried a bunch of times to get this to finish locally. It always seems to get hung up on a few of the getObjktById requests.

File: src/lib/conseil.js [hunk 4]
```diff
@@ -0,0 +1,1101 @@
+const conseiljs = require('conseiljs')
```
Comment (mikesizz) [inline src/lib/conseil.js:1]: Lots of undefined variables here. Needs to be cleaned up.

Comment (mikesizz) [reply src/lib/conseil.js:1]: image

Comment (mikesizz) [reply src/lib/conseil.js:1]: Getting same results as master when above is fixed.

File: src/lib/conseil.js (hunk 1 above)
Comment (mikesizz) [reply src/lib/conseil.js:771]: Seems like putting the earlier date first works (go figure). Not sure what the desired start date is as master is putting max_time (0 for /random) first.

**Comment by mikesizz** [comment]
Additional changes that should be included now or in the near future:
- Functional test suite
- Move configuration values to config.js with process.env overrides
  - Current values as defaults
- GitHub PR template
- CHANGELOG.md
- Clean up .env file usage
  - Add .env.sample for ppl to copy
- License file
- Improve linting rules
- Remove lambda configs if no longer needed

**Comment by tarwin** [comment]
Looks great to me. A lot easier to follow the code. (Haven't actually run it, just looked through each flow up to the `conseil` calls).

File: src/lib/conseil.js (hunk 4 above)
Comment (veqtor) [reply src/lib/conseil.js:1]: What does this mean? They need to be predefined?

File: src/lib/router/readHdaoFeed.js (hunk 3 above)
Comment (veqtor) [reply src/lib/router/readHdaoFeed.js:15]: I'd say if any of those fail, let them and ignore them

**Pull request merged**